packages/in_app_purchase/lib/src/in_app_purchase_connection/product.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// The class represents the product object, it combined the common fields in the [SkuDetailsWrapper] and [SKProductWrapper].
+///
+/// You can use this object for simple IAP operations. For more detailed and platform specific IAP operations,
+/// see [SkuDetailsWrapper] and [SKProductWrapper] instead.
+/// The programmer should not need to instantiate this object; rather use after returned by [InAppPurchaseConnection.getProductList].
+class Product {
```

Comment:
It's tricky but I think naming this just `Product` has the potential to be confusing with purchased products. This is the data structure representing store listings of available products to buy/product details/product information, not purchased products themselves. Maybe name this `ProductInformation`? `ProductDetails`?

Reply:
SG